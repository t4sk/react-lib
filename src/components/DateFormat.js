import React from "react"
import PropTypes from "prop-types"
import moment from "moment"

export function DateFormat(props) {
  const { format, date } = props

  return moment(date).format(format)
}

DateFormat.defaultProps = {
  format: "YYYY-MM-DD HH:mm:ss",
}

DateFormat.propTypes = {
  format: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired,
}

export default DateFormat
