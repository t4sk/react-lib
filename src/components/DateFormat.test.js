import React from "react"
import { shallow } from "enzyme"
import { DateFormat } from "./DateFormat"

const props = {
  date: new Date(0).toISOString(),
}

test("it renders", () => {
  const component = shallow(<DateFormat {...props} />)

  expect(component).toMatchSnapshot()
})
