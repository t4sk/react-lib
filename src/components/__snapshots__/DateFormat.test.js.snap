// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `"1970-01-01 09:00:00"`;
